collect_app/src/main/java/org/odk/collect/android/tasks/SaveToDiskTask.java

```diff
@@ -436,6 +453,20 @@ private void exportData(boolean markCompleted) throws IOException, EncryptionExc
             // if encrypted, delete all plaintext files
             // (anything not named instanceXml or anything not ending in .enc)
             if (isEncrypted) {
+                // Clear the geometry. Done outside of updateInstanceDatabase to avoid multiple
+                // branches and because it has no knowledge of encryption status.
+                ContentValues values = new ContentValues();
+                values.put(InstanceColumns.GEOMETRY, (String) null);
+                values.put(InstanceColumns.GEOMETRY_TYPE, (String) null);
+                try {
+                    int updated = Collect.getInstance().getContentResolver().update(uri, values, null, null);
+                    if (updated < 1) {
+                        Timber.w("Instance geometry not cleared after encryption");
+                    }
+                } catch (IllegalArgumentException e) {
```

Comment:
This could happen if the `uri` wasn't properly set to the instance's in one of the two cases where `FormEntryActivity` was called with a blank form URI.